### PR TITLE
CMake: Replace custom SPDLOG_BUILD_SHARED by standard BUILD_SHARED_LIBS

### PR DESCRIPTION
**Pull request by concatime**
PR for https://github.com/gabime/spdlog/issues/1848.
It works on GNU+Linux.
It should work on all supported platforms.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -124,12 +127,12 @@ if(NOT SPDLOG_FMT_EXTERNAL AND NOT SPDLOG_FMT_EXTERNAL_HO)
...
-        configure_file(${CMAKE_CURRENT_SOURCE_DIR}/cmake/version.rc.in ${CMAKE_CURRENT_BINARY_DIR}/version.rc @ONLY)
-        list(APPEND SPDLOG_SRCS ${CMAKE_CURRENT_BINARY_DIR}/version.rc)
-    endif()
-    add_library(spdlog SHARED ${SPDLOG_SRCS} ${SPDLOG_ALL_HEADERS})
+if(BUILD_SHARED_LIBS AND WIN32)
+    configure_file(${CMAKE_CURRENT_SOURCE_DIR}/cmake/version.rc.in ${CMAKE_CURRENT_BINARY_DIR}/version.rc @ONLY)
+    list(APPEND SPDLOG_SRCS ${CMAKE_CURRENT_BINARY_DIR}/version.rc)
+endif()
+add_library(spdlog ${SPDLOG_SRCS} ${SPDLOG_ALL_HEADERS})
+if (BUILD_SHARED_LIBS)
```
Comment (gabime) [inline CMakeLists.txt:135]: No need for any changes from orig code except for the ```if(SPDLOG_BUILD_SHARED OR BUILD_SHARED_LIBS) ```

File: CMakeLists.txt [hunk 2]
```diff
@@ -50,8 +50,11 @@ endif()
 
 option(SPDLOG_BUILD_ALL "Build all artifacts" OFF)
 
-# build shared option
-option(SPDLOG_BUILD_SHARED "Build shared library" OFF)
+# Retain backward compatibility
+# TODO: Remove in 2.x
+if (DEFINED SPDLOG_BUILD_SHARED)
+    option(BUILD_SHARED_LIBS "Build shared library" ${SPDLOG_BUILD_SHARED})
+endif()
```
Comment (gabime) [inline CMakeLists.txt:57]: No need to redefine BUILD_SHARED_LIBS. 

just change line line 127 to 
```if(SPDLOG_BUILD_SHARED OR BUILD_SHARED_LIBS)``` and we are done. Single line change is all this PR can reduced to.

**Review by gabime — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (concatime) [reply CMakeLists.txt:135]: Your proposition will not work. If I set `SPDLOG_BUILD_SHARED` to true w/o setting `BUILD_SHARED_LIBS` to true, `add_library` will still build a static library.

File: CMakeLists.txt (hunk 2 above)
Comment (concatime) [reply CMakeLists.txt:57]: Well, the idea was to use only one `add_library()`, which internally checks for `BUILD_SHARED_LIBS`.

**Comment by gabime** [comment]
The idea is to support BUILD_SHARED_LIBS, which can be achieved in single line :)

**Comment by concatime** [comment]
@gabime I thought the idea was to _use_, not _just support_, `BUILD_SHARED_LIBS`. I will update the code then.

**Comment by concatime** [comment]
I replaced https://github.com/concatime/spdlog/commit/5dc039f8ecab5b5e5c524d4d4c17c5b58fef1e02 by https://github.com/concatime/spdlog/commit/3eed64e5c4b2181349bae97462357051c18fb0ed.

**Review by gabime — APPROVED** [review]

**Comment by gabime** [comment]
Thanks @concatime.

**Pull request merged**